Frameworks/include/CGIWICBitmap.h

```diff
@@ -106,10 +115,31 @@ class CGIWICBitmap
     : public RuntimeClass<RuntimeClassFlags<RuntimeClassType::ClassicCom>, IAgileObject, FtmBase, IWICBitmap, ICGDisplayTexture> {
 public:
     CGIWICBitmap(_In_ IDisplayTexture* texture, _In_ WICPixelFormatGUID pixelFormat, _In_ UINT height, _In_ UINT width)
-        : m_texture(texture), m_pixelFormat(pixelFormat), m_height(height), m_width(width) {
+        : m_dataBuffer(nullptr), m_texture(texture), m_pixelFormat(pixelFormat), m_height(height), m_width(width), m_freeData(false) {
+    }
+
+    // if data is null, then a buffer that is the size of bytes per row is allocated.
+    CGIWICBitmap(_In_ void* data, _In_ WICPixelFormatGUID pixelFormat, _In_ UINT height, _In_ UINT width)
+        : m_texture(nullptr), m_pixelFormat(pixelFormat), m_height(height), m_width(width), m_freeData(false) {
+        // Obtain bytes per row from pixelFormat
+        const __CGImagePixelProperties* properties = _CGGetPixelFormatProperties(m_pixelFormat);
+        FAIL_FAST_IF(properties == nullptr);
+        m_bytesPerRow = (properties->bitsPerPixel >> 3) * m_width;
+
+        if (data) {
+            m_dataBuffer = static_cast<BYTE*>(data);
+        } else {
+            m_dataBuffer = static_cast<BYTE*>(IwMalloc(m_height * m_bytesPerRow));
+            m_freeData = true;
+        }
     }
 
     ~CGIWICBitmap() {
+        if (m_freeData) {
```

Comment:
The logic seems odd, given that naturally it's suggested that we allocate the data at most times.
Although RAII are great, I feel as if this is seems unconventional. 
If(free) {
 .release();
}

Also it makes the assignments make it look like, it could be free-ed.
I'm leaving it as is for now. 

---
In reply to: [88114811](https://github.com/Microsoft/WinObjC/pull/1371#discussion_r88114811) [](ancestors = 88114811,87863221,87861378)